internal/controller/controllers/agent_controller_test.go

```diff
@@ -335,14 +336,15 @@ var _ = Describe("agent reconcile", func() {
 				return c.Get(ctx, name, cd)
 			},
 		).Times(2)
-		mockClient.EXPECT().Update(gomock.Any(), gomock.AssignableToTypeOf(&v1beta1.Agent{})).DoAndReturn(
-			func(ctx context.Context, agent *v1beta1.Agent) error {
+		ctx_object := reflect.TypeOf((*context.Context)(nil)).Elem()
```

Comment:
You would have thought ``gomock.AssignableToTypeOf(context.TODO())`` will work, right? Wrong!
They also have this (ugly) specific usage in their documentation: https://pkg.go.dev/github.com/golang/mock/gomock#AssignableToTypeOf

Reply:
i would't check the context, it's a given

Reply:
for context just use `gomock.Any()`

Reply:
sure